src/array.jl

```diff
@@ -167,6 +167,41 @@ const copyfun = VERSION >= v"0.7.0-DEV.3057" ? :(copyto!) : :(copy!)
         Mem.transfer!(dst.buf, src.buf, length(src) * sizeof(T))
         return dst
     end
+
+    """
+        $copyfun{T}(dst::CuArray{T}, src::SubArray{T,N,A,I,true})
+
+    Copy an array view from a host array `src` to a device array `dst` in place. Both arrays
+    should have an equal length, and the stide of the view must be unity in all dimensions.
+    """
+    function Base.$copyfun(dst::CuArray{T}, src::SubArray{T,N,A,I,true}) where {T,N,A,I}
+        if length(dst) != length(src)
+            throw(ArgumentError("Inconsistent array length."))
+        end
+        if any(strides(src) .!= 1)
```

Comment:
This will fail for any multidimensional array, where even a dense array has a stride bigger than one along higher dimensions.

`strides` is not entirely safe yet, but see https://github.com/JuliaLang/julia/pull/25321 (which should make it safe). If you rely on an unsafe `strides`, you might get wrong answers or segfaults.

Reply:
@timholy good point, thanks. I was concentrating on the particular problem I have in #76 and hadn't properly considered this. 

I could fix this to work for views which only restrict the trailing dimension to a contiguous region, but that's too peculiar. Alternatively, I could restrict the method only to operate on views which result in a vector, which would still be useful. 

That said, if `strides` isn't safe, then that might be a showstopper. Feel free to close the PR if that's the case - I can always implement these methods privately in my package until things stabilise.